src/parsers/urdf_yaml_conv.cpp

```diff
@@ -0,0 +1,135 @@
+#include <RBDyn/parsers/urdf.h>
+#include <RBDyn/parsers/yaml.h>
+
+#include <fstream>
+#include <iostream>
+
+std::string getContent(std::istream & in)
+{
+  // Without sync_with_stdio = false, in_avail always returns zero even if data is available
+  std::ios::sync_with_stdio(false);
+  const auto available_char = in.rdbuf()->in_avail();
+  std::ios::sync_with_stdio(true);
+  if(available_char == 0)
+  {
+    return {};
+  }
+
+  std::string content;
+  std::string line;
+  while(std::getline(in, line))
+  {
+    content += line + '\n';
+  }
+  return content;
+}
+
+int main(int argc, const char * argv[])
+{
+  auto print_usage = [&]() {
+    std::cout << "Converts between URDF and YAML robot model formats. The input format is detected based on the given "
+                 "content.\n";
+    std::cout << "Usage:\n";
+    std::cout << "\t" << argv[0] << ": reads from the standard input and prints to the standard output\n";
+    std::cout << "\t" << argv[0] << " input_file: reads input_file and prints to the standard output\n";
+    std::cout << "\t" << argv[0] << " input_file output_file: reads input_file and writes to output_file\n";
+  };
+  if(argc == 2 && (std::strcmp(argv[1], "-h") == 0 || std::strcmp(argv[1], "--help") == 0))
+  {
+    print_usage();
+    std::exit(0);
+  }
+
+  if(argc > 3)
+  {
+    print_usage();
+    std::exit(-1);
+  }
+
+  const auto stdin_content = getContent(std::cin);
+  const auto file_input = stdin_content.size() == 0;
+  const auto file_output = (file_input && argc == 3) || (!file_input && argc == 2);
```

Comment:
I'm a little confused here about how it works with the standard input?

Reply:
`getContent` tries to read everything available on the given input stream, either `std::cin` or a file. So if `stdin_content` is empty, it means that there's nothing to read from `stdin` and that the input must be given as a file. Then the output could be either `stdout` or a file and that depends on both the type of input and number of arguments.

So in the end the program can be called like this:
1. `urdf_yaml_converter` : reads from `stdin` and writes to `stdout`
2. `urdf_yaml_converter out.ext` : reads from `stdin` and writes to `out.ext`
3. `urdf_yaml_converter in.ext` : reads from `in.ext` and writes to `stdout`
4. `urdf_yaml_converter in.ext out.ext` : reads from `in.ext` and writes to `out.ext`

Reply:
Thanks for the clarification, but how do you distinguish case 2 and 3?

Reply:
It depends if there is data available on `stdin` or not.

Reply:
So, typically for 2
```bash
$ cat in.urdf | urdf_yaml_converter out.yaml
```
And for 3:
```bash
$ urdf_yaml_converter in.urdf
```
right?

Reply:
Exactly

Reply:
Just realized that 2. is not explained in the usage, I'll update it